devicetypes/smartthings/zigbee-power-meter.src/zigbee-power-meter.groovy

```diff
@@ -12,98 +12,107 @@
-        
-    }
+		fingerprint profileId: "0104", deviceId:"0053", inClusters: "0000, 0003, 0004, 0B04, 0702", outClusters: "0019", manufacturer: "", model: "E240-KR080Z0-HA", deviceJoinName: "Smart Sub-meter (CT Type)"
+	}
 
-    // tile definitions
-    tiles(scale: 2) {
-        multiAttributeTile(name:"power", type: "generic", width: 6, height: 4){
-            tileAttribute("device.power", key: "PRIMARY_CONTROL") {
-                attributeState("default", label:'${currentValue} W')
-            }
-            tileAttribute("device.energy", key: "SECONDARY_CONTROL") {
-                attributeState("default", label:'${currentValue} kWh')
-            }
-        }
-        standardTile("reset", "device.energy", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
-            state "default", label:'reset kWh', action:"reset"
-        }
-        standardTile("refresh", "device.power", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
-            state "default", label:'', action:"refresh.refresh", icon:"st.secondary.refresh"
-        }
+	// tile definitions
+	tiles(scale: 2) {
+		multiAttributeTile(name:"power", type: "generic", width: 6, height: 4){
+			tileAttribute("device.power", key: "PRIMARY_CONTROL") {
+				attributeState("default", label:'${currentValue} W')
+			}
+			tileAttribute("device.energy", key: "SECONDARY_CONTROL") {
+				attributeState("default", label:'${currentValue} kWh')
+			}
+		}
+		standardTile("reset", "device.energy", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", label:'reset kWh', action:"reset"
+		}
+		standardTile("refresh", "device.power", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", label:'', action:"refresh.refresh", icon:"st.secondary.refresh"
+		}
 
-        main (["power", "energy"])
-        details(["power", "energy", "reset", "refresh"])
-    }
+		main (["power", "energy"])
+		details(["power", "energy", "reset", "refresh"])
+	}
 }
 
+def getATTRIBUTE_READING_INFO_SET() { 0x0000 }
+def getATTRIBUTE_HISTORICAL_CONSUMPTION() { 0x0400 }
+
 def parse(String description) {
-    log.debug "description is $description"
-    def event = zigbee.getEvent(description)
-    if (event) {
-        log.info event
-        sendEvent(event)
-    } else {
-        List result = []
-        def descMap = zigbee.parseDescriptionAsMap(description)
-        log.debug "Desc Map: $descMap"
-                
-        List attrData = [[clusterInt: descMap.clusterInt ,attrInt: descMap.attrInt, value: descMap.value]]
-        descMap.additionalAttrs.each {
-            attrData << [clusterInt: descMap.clusterInt, attrInt: it.attrInt, value: it.value]
+	log.debug "description is $description"
+	def event = zigbee.getEvent(description)
+	if (event) {
+		log.info "event enter:$event"
+        if(event.name=="power"){
```

Comment:
Please use tabs instead of spaces and add spaces in between the different tokens. For example:

```
		log.debug "event enter:$event"
		if (event.name == "power") {
			event.value = event.value / 1000
		} else if (event.name == "energy") {
			event.value = event.value / 1000000
		}
		log.debug "event outer:$event"
```

Also, please use log.debug instead of log.info, as log.info (especially for frequent events) generates a lot of extra data.